lua/entities/gmod_tardis/modules/cl_classic_doors.lua

```diff
@@ -0,0 +1,29 @@
+ENT:AddHook("ShouldDraw", "classic_doors_support", function(self)
+	if self.metadata.EnableClassicDoors ~= true then return end
+
+	local ply = LocalPlayer()
+
+	local inside = (ply:GetTardisData("exterior") == self
+		and not ply:GetTardisData("thirdperson")
+		and not ply:GetTardisData("destination"))
+
+	if inside then return false end
+
+end)
+
+ENT:AddHook("ShouldDrawPart", "classic_doors_support", function(self, part)
+	if self.metadata.EnableClassicDoors ~= true then return end
+	if part == nil then return end
+
+	local ply = LocalPlayer()
+
+	local inside = (ply:GetTardisData("exterior") == self
```

Comment:
We should not be checking player state as this will quickly break down if the player is using a camera view or something else. We should be using wp.drawing + wp.drawingent to determine if we're currently drawing the appropriate portal and hide it in that context instead. Same with the other instances where we are checking these variables.

Reply:
Should be resolved